sechub-scan/src/main/java/com/mercedesbenz/sechub/domain/scan/product/config/ProfileMessageHandler.java

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.domain.scan.product.config;
+
+import static com.mercedesbenz.sechub.sharedkernel.messaging.MessageDataKeys.PROJECT_ASSIGNED_PROFILE_IDS;
+import static com.mercedesbenz.sechub.sharedkernel.messaging.MessageDataKeys.PROJECT_IDS;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Component;
+
+import com.mercedesbenz.sechub.sharedkernel.messaging.DomainMessage;
+import com.mercedesbenz.sechub.sharedkernel.messaging.DomainMessageSynchronousResult;
+import com.mercedesbenz.sechub.sharedkernel.messaging.IsRecevingSyncMessage;
+import com.mercedesbenz.sechub.sharedkernel.messaging.MessageID;
+import com.mercedesbenz.sechub.sharedkernel.messaging.SynchronMessageHandler;
+
+@Component
+public class ProfileMessageHandler implements SynchronMessageHandler {
+
+    private final ProductExecutionProfileRepository productExecutionProfileRepository;
+
+    public ProfileMessageHandler(ProductExecutionProfileRepository productExecutionProfileRepository) {
+        this.productExecutionProfileRepository = productExecutionProfileRepository;
+    }
+
+    @IsRecevingSyncMessage(MessageID.REQUEST_PROFILE_IDS_FOR_PROJECT)
+    @Override
```

Comment:
Please take a look into `ScanMessageHandler.java`:
I designed the synchronous messaging in a way, 
that we have always two different message ids: one for the question, another one for the answer.

It is interesting that the mechanism works with same message id inside the result.
But when it comes to generartion of diagrams for events, I think this could become problematic.

I would suggest to introduce an additional message id `RESULT_PROFILE_IDS_FOR_PROJECT`
You can of course use the same data type inside the new message Id.

Also you have to markup that your are answering here:
```suggestion
@IsSendingSyncMessageAnswer(value = MessageID.RESULT_PROFILE_IDS_FOR_PROJECT, answeringTo = MessageID.REQUEST_PROFILE_IDS_FOR_PROJECT)
    @Override
```